lib/services/elasticsearch.js

```diff
@@ -1705,43 +1706,43 @@ class ElasticSearch extends Service {
     }
     /* end critical code section */
 
-    return this.mGet(index, collection, validIds)
-      .then(({ items }) => {
-        let idx = 0;
+    const {items} = await this.mGet(index, collection, validIds);
 
-        /**
-         * @warning Critical code section
-         *
-         * request can contain more than 10K elements
-         */
-        for (let i = 0; i < validIds.length; i++) {
-          const
-            validId = validIds[i],
-            item = items[idx];
+    let idx = 0;
 
-          if (item && item._id === validId) {
-            query.ids.values.push(validId);
-            idx++;
-          }
-          else {
-            partialErrors.push({
-              _id: validId,
-              reason: 'document not found',
-              status: 404
-            });
-          }
-        }
-        /* end critical code section */
+    /**
+     * @warning Critical code section
+     *
+     * request can contain more than 10K elements
+     */
+    for (let i = 0; i < validIds.length; i++) {
+      const
+        validId = validIds[i],
+        item = items[idx];
 
-        // @todo duplicated query to get documents body, mGet here and search in
-        // deleteByQuery
-        return this.deleteByQuery(index, collection, query, { refresh, timeout });
-      })
-      .then(({ documents }) => ({
-        documents,
-        errors: partialErrors
-      }))
-      .catch(error => this._esWrapper.reject(error));
+      if (item && item._id === validId) {
+        query.ids.values.push(validId);
+        idx++;
+      }
+      else {
+        partialErrors.push({
+          _id: validId,
+          reason: 'document not found',
+          status: 404
+        });
+      }
+    }
+    /* end critical code section */
+
+    // @todo duplicated query to get documents body, mGet here and search in
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)